File: app/src/main/java/com/lis/player_java/data/room/model/MusicArtists.java
package com.lis.player_java.data.room.model;

import androidx.room.Embedded;
import androidx.room.Relation;

import java.util.List;

public class MusicArtists {
    @Embedded public MusicDB musicDB;
    @Relation(
            parentColumn = "id",
            entityColumn = "musicId"
    )
    public List<ArtistDB> artists;
    public ArtistType type;
}

File: app/src/main/java/com/lis/player_java/data/room/model/MusicArtists.kt
package com.lis.player_java.data.room.model

import androidx.room.Embedded
import androidx.room.Relation

data class MusicArtists(
    @Embedded
    val musicDB: MusicDB,
    @Relation(parentColumn = "id", entityColumn = "musicId")
    val artists: List<ArtistDB>,
    val type: ArtistType
)